ads_common/lib/ads_common/auth/oauth2_handler.rb

```diff
@@ -91,7 +91,10 @@ def get_token(credentials = nil, force_refresh = false)
 
       # Refreshes access token from refresh token.
       def refresh_token!()
-        return nil if @token.nil? or @token[:refresh_token].nil?
+        if @token.nil? or @token[:refresh_token].nil?
+          raise AuthError.new("Unable to refresh token due to missing oauth information.  Check account settings.")
```

Comment:
so the conditional in `get_token` is determining if we need to refresh the token we have.  that conditional will not assign the value `nil` to token, it's just an optional override, which when triggered and `@token` or `@refresh_token` don't exist, token gets overridden to `nil`.  Raising here seems to be the pinch point for refreshing tokens and previously returning `nil`.